imports/plugins/core/ui/client/components/calendarPicker/calendarPicker.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { registerComponent } from "@reactioncommerce/reaction-components";
+import { DayPickerRangeController } from "react-dates";
+import omit from "lodash/omit";
+
+class CalendarPicker extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.initialStartDate,
+      endDate: props.initialEndDate,
+      focusedInput: props.autoFocusEndDate ? "endDate" : "startDate"
+    };
+  }
+
+  onDatesChange = ({ startDate, endDate }) => {
+    this.setState({
+      startDate,
+      endDate
+    });
+
+    if (this.props.onDatesChange) {
+      this.props.onDatesChange(startDate, endDate);
```

Comment:
Why are we wrapping this in a conditional? Will this function not always exist? And if not shouldn't we throw an error? I feel like this is an anti-pattern that has crept into our code somehow.

Also, maybe this is unavoidable but it seems like having this function and the one passed in via props both having the same name is a recipe for a bug down the road?

Reply:
The reason we have that conditional is b/c you dont have to have a prop function for when dates change; but in case you want to add some operations/ or change some stuff onDateChange specific to the component that extends the calendarPicker you could do that under this function.
Other example: https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/core/ui/client/components/menu/menu.js#L7

Reply:
Ok, makes sense to me